snuba/utils/streams/abstract.py

```diff
@@ -138,6 +138,24 @@ def poll(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def tell(self) -> Mapping[TStream, TOffset]:
+        """
+        Return the read offsets for all assigned streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def seek(self, offsets: Mapping[TStream, TOffset]) -> None:
+        """
+        Change the read offsets for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        raise NotImplementedError
```

Comment:
This should also document what happens when an unassigned `TStream` is provided.

Reply:
Should we specify an exception we would throw if the requested offset is not reachable. Like if it is too old and not in the retention window or if it is in the future ?

Reply:
I'm torn between doing that here or allowing out of range offsets to be set here and erroring on `poll`.

There's no guarantee that an offset that is available when it is seeked to will still be available when it's attempted to be read, so `poll` will be able to raise in this situation regardless, and using `get_watermark_offsets` here will incur a network IO hit, or potentially read stale high/low watermarks from cache.